hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/ReconServer.java

```diff
@@ -144,6 +155,50 @@ public void join() {
     }
   }
 
+  /**
+   * Logs in the Recon user if security is enabled in the configuration.
+   *
+   * @param conf OzoneConfiguration
+   * @throws IOException, AuthenticationException in case login fails.
+   */
+  private static void loginReconUserIfSecurityEnabled(OzoneConfiguration  conf)
+      throws IOException, AuthenticationException {
+    if (OzoneSecurityUtil.isSecurityEnabled(conf)) {
+      loginReconUser(conf);
+    }
+  }
+
+  /**
+   * Login Recon service user if security is enabled.
+   *
+   * @param  conf OzoneConfiguration
+   * @throws IOException, AuthenticationException
+   */
+  private static void loginReconUser(OzoneConfiguration conf)
+      throws IOException, AuthenticationException {
+
+    if (SecurityUtil.getAuthenticationMethod(conf).equals(
+        UserGroupInformation.AuthenticationMethod.KERBEROS)) {
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("Ozone security is enabled. Attempting login for Recon user. "
```

Comment:
NIT: this can be a LOG.info() like many other hadoop daemons. It can help identifying misconfigured principal/keytab. .